src/App.jsx

```diff
@@ -14,45 +16,128 @@ export const goodsFromServer = [
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPreparedArray(goods, { sortField, reversed }) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods = preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          if (good1 > good2) {
+            return 1;
+          }
+
+          if (good1 < good2) {
+            return -1;
+          }
+
+          return 0;
+
+        case SORT_FIELD_LENGTH:
+          if (good1.length > good2.length) {
+            return 1;
+          }
+
+          if (good1.length < good2.length) {
+            return -1;
+          }
+
+          return 0;
+
+        default: return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedArray(
+    goodsFromServer,
+    { sortField, reversed: isReversed },
+  );
+
+  const reset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames(
+            'button is-info',
+            {
+              'is-light': sortField !== SORT_FIELD_NAME,
+            },
+          )}
+          onClick={() => setSortField(SORT_FIELD_NAME)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={classNames(
+            'button is-success',
+            {
+              'is-light': sortField !== SORT_FIELD_LENGTH,
+            },
+          )}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={classNames(
+            'button is-warning',
+            {
+              'is-light': !isReversed,
+            },
+          )}
+          onClick={() => setIsReversed(current => !current)}
+        >
+          Reverse
+        </button>
+
+        {(sortField !== '' || isReversed !== false)
```

Comment:
the main idea of the related components was the following: you shouldn't compare the boolean variable with `false` and the string variable with an empty string: [conditional rendering](https://react.dev/learn/conditional-rendering#logical-and-operator-)
Empty string is a [falsy](https://www.freecodecamp.org/news/falsy-values-in-javascript/) value